// Generated by sylk.build, DO NOT EDIT.
syntax = "proto3";
package sylklabs.iot.v1;
option go_package = "github.com/sylk-build/sylk-examples/sylk-iot/services/protos/sylklabs/iot/v1;iotv1";

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

service IotService {
	// [sylklabs.iot.v1.IotService.Subscribe] - 
	rpc Subscribe (sylklabs.iot.v1.Topic) returns (stream sylklabs.iot.v1.SensorData);
	// [sylklabs.iot.v1.IotService.Publish] - 
	rpc Publish (sylklabs.iot.v1.PublishRequest) returns (google.protobuf.Empty);
}


// [sylklabs.iot.v1.PublishRequest] - 
message PublishRequest {
	// [sylklabs.iot.v1.PublishRequest.topic] - 
	string topic = 1;
	// [sylklabs.iot.v1.PublishRequest.data] - 
	sylklabs.iot.v1.SensorData data = 2;
}

// [sylklabs.iot.v1.Topic] - The Topic message represents the topic to which the IoT server and Grafana plugin will subscribe and publish. It will have a single field called name, which is a string representing the name of the topic.
message Topic {
	// [sylklabs.iot.v1.Topic.name] - 
	string name = 1;
}

// [sylklabs.iot.v1.SensorData] - 
message SensorData {
	// [sylklabs.iot.v1.SensorData.sensor_id] - 
	string sensor_id = 1;
	// [sylklabs.iot.v1.SensorData.value] - 
	double value = 2;
	// [sylklabs.iot.v1.SensorData.timestamp] - 
	google.protobuf.Timestamp timestamp = 3;
}